src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs

```diff
@@ -186,6 +186,12 @@ internal sealed partial class ConsoleHost
 
                 s_cpp.Parse(tempArgs);
 
+#if UNIX
+                // On Unix, logging has to be deferred until after command-line parsing
+                // completes to allow overriding logging options.
+                PSEtwLog.LogConsoleStartup();
+#endif
```

Comment:
😕 Is there no way to unify this for all platforms?

Reply:
Yes it is but I'll leave it open for feedback. I don't want.to change the Windows behavior without a little thought. The non-Windows code has to change.

Reply:
I think we should do the config file overriding through an environment variable until the command-line parsing has been moved earlier.

Reply:
I think we should do the config file overriding through an environment variable until the command-line parsing has been moved earlier. Then this `#if UNIX` block won't be needed.

Reply:
It's still open.  Experimental feature has it as a pending design issue.

Reply:
Does it actually make sense to have all this output getting sent to the logging process if someone is running Powershell interactively?

Perhaps a command line option to tell Powershell that it's being run as a non-interactive or batch process and then logging of everything or anything above say "informational" could be sent to "syslog" via the standard "C" syslog functions.

That anything above "informational" option could actually be a configuration added to the system file "syslog.conf" or directory "syslog.d" to limit the output at that point instead of doing it inside Powershell.  I currently do this myself to get rid of all the chatter from Powershell as follows:

    # Only log Powershell message above Informational level
    :programname,startswith,"powershell" stop

Obviously this rule is dropping everything sent to the logger by Powershell.  You probably don't want this but I'm not interested in any of the logging from it at the moment and using complex regular expression logging rules can slow down the logger.

Reply:
@vramm You can control the logging level for PowerShell in powershell.config.json.  Take a look at 'Configuring Logging on non-Windows systems' in [about_logging](https://github.com/PowerShell/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Logging.md)

As far as interactive versus batch/non-interactive, there is no concept in PowerShell from a logging perspective.  However, you can start up PowerShell for your interactive sessions using -settingsFile to accomplish what you are looking for.